src/argilla/client/feedback/dataset/remote.py

```diff
@@ -0,0 +1,257 @@
+    from argilla.client.feedback.types import AllowedFieldTypes, AllowedQuestionTypes
+    from argilla.client.sdk.v1.datasets.models import FeedbackItemModel
+    from argilla.client.workspaces import Workspace
+
+
+warnings.simplefilter("always", DeprecationWarning)
+
+
+class RemoteFeedbackRecords:
+    def __init__(self, client: "httpx.Client", id: "UUID", questions: List["AllowedQuestionTypes"]) -> None:
+        self.client = client
+        self.id = id
+
+        self.__question_id2name = {question.id: question.name for question in questions}
+        self.__question_name2id = {value: key for key, value in self.__question_id2name.items()}
+
+    def __repr__(self) -> str:
+        return (
+            "The `records` of a `FeedbackDataset` in Argilla are being lazily"
+            " fetched, and never stored locally. You can either loop over `records`"
+            " or access them by index, and those will be fetched from Argilla on the"
+            " fly."
+        )
+
+    def __parse_record(self, record: "FeedbackItemModel") -> RemoteFeedbackRecord:
+        record = record.dict(
+            exclude={
+                "inserted_at": ...,
+                "updated_at": ...,
+                "responses": {"__all__": {"id", "inserted_at", "updated_at"}},
+                "suggestions": {"__all__": {"id"}},
+            },
+            exclude_none=True,
+        )
+        for suggestion in record.get("suggestions", []):
+            suggestion.update({"question_name": self.__question_id2name[suggestion["question_id"]]})
+        return RemoteFeedbackRecord(client=self.client, name2id=self.__question_name2id, **record)
+
+    def __len__(self) -> int:
+        try:
+            response = datasets_api_v1.get_metrics(client=self.client, id=self.id)
+        except Exception as e:
+            raise Exception(
+                f"Failed while getting the metrics from the current `FeedbackDataset` in Argilla with exception: {e}"
+            ) from e
+        return response.parsed.records.count
+
+    def __getitem__(self, key: Union[slice, int]) -> Union[RemoteFeedbackRecord, List[RemoteFeedbackRecord]]:
+        offsets = []
+        limit = None
+        if isinstance(key, slice):
+            start, stop, step = key.indices(len(self))
+            if step is not None and step != 1:
+                return [self[i] for i in range(start, stop, step)]
+            if start < 0:
+                start += len(self)
+            if stop < 0:
+                stop += len(self)
+            if start < 0 or stop < 0:
+                raise IndexError("Index out of range")
+            limit = stop - start
+            offsets = [start] if limit < FETCHING_BATCH_SIZE else list(range(start, stop, FETCHING_BATCH_SIZE))
+        elif isinstance(key, int):
+            if key < 0:
+                key += len(self)
+            if key < 0 or key >= len(self):
+                raise IndexError("Index out of range")
```

Comment:
Just to avoid several calls to compute the lenght
```suggestion
        offsets = []
        limit = None
        records_lenght= len(self)
        if isinstance(key, slice):
            start, stop, step = key.indices()
            if step is not None and step != 1:
                return [self[i] for i in range(start, stop, step)]
            if start < 0:
                start += records_lenght
            if stop < 0:
                stop += records_lenght
            if start < 0 or stop < 0:
                raise IndexError("Index out of range")
            limit = stop - start
            offsets = [start] if limit < FETCHING_BATCH_SIZE else list(range(start, stop, FETCHING_BATCH_SIZE))
        elif isinstance(key, int):
            if key < 0:
                key += records_lenght
            if key < 0 or key >= records_lenght:
                raise IndexError("Index out of range")
```

Reply:
Thanks makes sense!